### ISSUE TITLE
File History

### ISSUE THREAD

**Issue by nunoperalta** [desktop/desktop#11990]
### Describe the feature or problem you’d like to solve

I cannot easily view the history of a specific file.

### Proposed solution

I'd like to be able to filter History by a specific file, similarly to this page:
https://github.com/desktop/desktop/commits/development/package.json

### Additional context

(just ideas)

I was thinking we could put the file name or path in a Search bar (which currently has "Select branch to compare...")

![image](https://user-images.githubusercontent.com/487079/114471092-f11ea480-9be7-11eb-9fc6-6e7c1692e10c.png)

and this would filter commits that contain a change to that file.

We could also have a new option in the Context Menu of the files changed on a commit:

![image](https://user-images.githubusercontent.com/487079/114470964-b0bf2680-9be7-11eb-9456-a52b5eb30a13.png)

which would put the file name/path in that Search bar above.

**Comment by tidy-dev** [comment]
@nunoperalta Thank you for the suggestion. This is not on our roadmap at this time, but I will mark it an enhancement for future consideration.

**Comment by maifeeulasad** [comment]
Can I implement this ? @tidy-dev

**Comment by tidy-dev** [comment]
@maifeeulasad Changing something like that search is fairly core to the history tab so would warrant quite a bit of discussion from our team and is not on our roadmap at this time. If you are interested in helping out on something, reviewing the issues with the help-wanted tag would be a good place to look. https://github.com/desktop/desktop/labels/help%20wanted

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/13110.

**Comment by robotboy655** [comment]
How is it possible that this basic feature is still missing? Simple option in a file's context menu would be sufficient.

**Comment by nunoperalta** [comment]
Yeah - the way it's done in VSCode/GitLens is a bit hard to use, as it's not straight forward to view the changes for a specific commit (it opens many new tabs with compares).

Having to log into Gitea (or whatever other UI website we use) is also not always convenient.

GitHub does it really nicely, having the history on the left and the changes on the right, and we just keep moving down as we review each commit.

This feature would greatly improve my experience.

**Comment by Post169** [comment]
Yes, it would be very wonderful if I could type in the name of a file and see the history of all the commits which modified it, browse through them, and find exactly when a certain change was made.

**Comment by uahgrad05** [comment]
I think tortoiseGit already has some functionality in place that does this. It would be nice to have that functionality in GithubDesktop as well.

**Comment by 0xdevalias** [comment]
> I was thinking we could put the file name or path in a Search bar (which currently has "Select branch to compare...")
> 
> ![image](https://user-images.githubusercontent.com/487079/114471092-f11ea480-9be7-11eb-9fc6-6e7c1692e10c.png)
> 
> and this would filter commits that contain a change to that file.

See also, this related issue:

- https://github.com/desktop/desktop/issues/6407

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/18411.

**Comment by johndpope** [comment]
<img width="534" alt="Screenshot 2024-04-12 at 11 43 19 am" src="https://github.com/desktop/desktop/assets/289994/3ed520bb-554a-4ab0-83f2-704b59046307">

https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens

31,000,000 installs!!!!
I'm pretty sure they're installing only because of this missing feature. 
please implement - or buy them out - and use their code. or hire more people to fix this. 
whatever it takes.
<img width="988" alt="Screenshot 2024-04-12 at 11 44 01 am" src="https://github.com/desktop/desktop/assets/289994/cf9b5d74-e1b5-4a3a-b1f5-5a5311457021">

**Comment by steveward** [comment]
Requested in #19094:

>it would be nice to see an option ,when we right click the file and as the popup menu appears , see an option "compare.." or "see differences." , and when we click compare ,there could be list of old history(revisions) of the file. and when i select two revisions from the list it compares selected two revisions. i could compare a revision with the latest revision or sometimes i could compare two older revisions i selected.
